binstar_client/inspect_package/pypi.py

```diff
@@ -270,7 +278,10 @@ def inspect_pypi_package_egg(filename, fileobj):
     if data is None:
         raise errors.NoMetadataError("Could not find EGG-INFO/PKG-INFO file in pypi sdist")
 
-    attrs = dict(Parser().parsestr(data).items())
+    if python_version_check()==3:
+        attrs = dict(Parser().parsestr(data).items())
+    else:
+        attrs = dict(Parser().parsestr(data.encode("UTF-8", "replace")).items())
```

Comment:
both fixes looks pretty much the same. Have you considered extracting them into a function? Maybe we can reduce some duplicated code